.github/actions/setup-echo-server/action.yml

```diff
@@ -0,0 +1,91 @@
+name: setup-echo-server
+decription: |
+  This action sets up an echo server associating a name to it.
+  The name, e.g. `httpbin.org`, will be resolved to 127.0.0.1 by dnsmasq.
+
+inputs:
+  ghcr_username:
+    required: true
+    type: string
+  ghcr_password:
+    required: true
+    type: string
+  server_name:
+    description: 'A record that dnsmasq will resolve to 127.0.0.1, e.g.: httpbin.org'
+    required: false
+    type: string
+    default: 'httpbin.org'
+  dns_port:
+    description: 'Port dnsmasq will listen to'
+    required: false
+    type: string
+    default: '8053'
+  upstream_dns_server:
+    description: 'DNS server used by dnsmasq to resolve entries not yet cached'
+    required: false
+    type: string
+    default: '1.1.1.1'
+
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Setup deps - apt-get'
+      shell: bash
+      run: |
+        # TODO: avoid Docker install on each run (use an image or an existing github action)
```

Comment:
What is the motivation for installing docker here when the runner already has it? Do we have plans for sharing this action soon?

Reply:
This was a quick hack to work on act. So long as it works on GitHub and act (for debugging, that's all we need).